installers/vagrant-dev/radiasoft-download.sh

```diff
@@ -171,7 +181,7 @@ EOF
     fi
     vagrant ssh <<EOF
 $(install_vars_export)
-curl $(install_depot_server)/index.sh | bash -s redhat-dev
+curl $(install_depot_server)/index.sh | bash -s redhat-dev "${vagrant_dev_no_nfs_src:+1}"
```

Comment:
When crossing script boundaries use variables, not anonymous args (home-env/install.sh was a mistake). `redhat_dev_git_nfs` would be the variable, but I don't think this should run here. This is a vagrant-dev issue. It will not happen in docker which also uses redhat-dev. 

This doesn't allow a cli override. Do it like the other args, and maybe inside vagrant_dev_main.

I would define it is `vagrant_dev_git_nfs` (affirmative) so it would default to true unless set in the environment. This might need a version check on the git as this wasn't always there so that it works on centos 7.

Reply:
> When crossing script boundaries use variables, not anonymous args

Will do.

>  but I don't think this should run here. This is a vagrant-dev issue

The problem is it needs to run after we have git installed (comes from redhat-base) but before we do any git operations in ~/src (ex `install_repo_eval home`). So, it can't be run in vagrant-dev w/o installing git in vagrant-dev and running before redhat-dev. Or breaking up redhat-dev and calling the pieces individually in vagrant-dev. What do you think would be best?

> It will not happen in docker which also uses redhat-dev.

Can you show me where that is? I searched and couldn't find the use https://github.com/search?type=code&auto_enroll=true&q=org%3Aradiasoft+redhat-dev

Reply:
> The problem is it needs to run after we have git installed 

This is subtle in a number of ways.

There is a conflict with install_repo_eval, which installs `~/.git_config` only if it doesn't already exist. However, it will exist as soon as `git config --global --add safe.directory '*'` is called.

I think home-env actually is ok, because all it does is clone, which doesn't need safe.directory.  This works:

```sh
sudo chmod 775 .
sudo chown root .
git clone https://github.com/biviosoftware/home-env
```

This does not work:

```sh
cd home-env
sudo chmod 775 .
sudo chown root .
git pull
fatal: detected dubious ownership in repository at '/home/vagrant/src/biviosoftware/home-env'
```

So, I think we are ok to leave it in vagrant-dev. Once vagrant-dev happens, it will be configured.

For idempotency, you could call it twice inside vagrant-dev: before redhat-dev and after. The function should check for the existence of `~/.gitconfig` and do nothing if it is not there.

> # POSIT: redhat-base installs git
> redhat_dev_fix_git_nfs_permissions ${1:+1}

This does not need to be posited. POSIT is only for cases where the assumption is not asserted by the code. `git` is called in redhat_dev_fix_git_nfs_permissions so it will fail immediately.

Bash functions should declare their arguments so the use of $1 should occur in the first line of this function.

You should document why this is necessary and how it couples with vagrant-dev.

Does the fix need to happen for the root user's gitconfig? I don't think that's right, because it really is a problem if root is in an arbitrary directory so I would say no, and I would document it as such.

While it is problematic for vagrant as well, it should be clearly documented that we consider it safe since this is a development VM. If the VM is compromised, it doesn't need to be done via git for a VM that has no sudo password for the user vagrant...

Reply:
> There is a conflict with install_repo_eval, which installs ~/.git_config

Yes, I meant to leave a comment about this. That is why I do the add after [the copy from the host](https://github.com/radiasoft/download/blob/a2e71aaa4e4f5238b448d6e85d1cebff269975c6/installers/vagrant-dev/radiasoft-download.sh#L175). But, I couldn't find an easy way to resolve with home-env...

> I think home-env actually is ok, because all it does is clone, which doesn't need safe.directory

I have ~/src/biviosoftware/home-env on my host machine. It get's mounted into the vm. So install.sh [does a pull](https://github.com/biviosoftware/home-env/blob/06c2f9b7bd1a6e54823df616406e0f7bfbcee0b9/install.sh#L45) which runs into the issue.

> For idempotency, you could call it twice inside vagrant-dev: before redhat-dev and after.

There is a lot going on here so I could be lost. But, git doesn't exist before redhat-dev so I don't think we can call it?

> Does the fix need to happen for the root user's gitconfig?

We don't mount anything under /root so we won't run into any problems. I'll make a note.

> it doesn't need to be done via git for a VM that has no sudo password for the user vagrant

lol, agreed.

Reply:
We had a conversation about this. I think we have a plan now.

Reply:
a5ce25c